packages/fmg-core/test/state.ts

```diff
@@ -25,85 +29,89 @@ contract('State', (accounts) => {
   });
 
   it("extracts the channelType", async () => {
-    let result = await stateLib.channelType.call(statePacket);
+    const result = await stateLib.channelType.call(statePacket);
     assert.equal(channelType, result);
   });
 
   it("extracts the channelNonce", async () => {
-    let result = await stateLib.channelNonce.call(statePacket);
+    const result = await stateLib.channelNonce.call(statePacket);
     assert.equal(channelNonce, result);
   });
 
   it("extracts the turnNum", async () => {
-    let result = await stateLib.turnNum.call(statePacket);
+    const result = await stateLib.turnNum.call(statePacket);
     assert.equal(turnNum, result);
   });
 
   it("extracts the number of participants", async () => {
-    let n = await stateLib.numberOfParticipants.call(statePacket);
+    const n = await stateLib.numberOfParticipants.call(statePacket);
     assert.equal(n, 2);
   });
 
   it("extracts the participants", async () => {
-    let result = await stateLib.participants.call(statePacket);
+    const result = await stateLib.participants.call(statePacket);
     assert.deepEqual(participants, result);
   });
 
   it("extracts the resolution", async () => {
-    let result = await stateLib.resolution.call(statePacket);
+    const result = await stateLib.resolution.call(statePacket);
     assert.equal(resolution[0].toNumber(), result[0]);
     assert.equal(resolution[1].toNumber(), result[1]);
   });
 
   it("identifies the mover based on the turnNum", async () => {
-    let mover = await stateLib.mover.call(statePacket);
+    const mover = await stateLib.mover.call(statePacket);
     // our state nonce is 15, which is odd, so it should be participant[1]
     assert.equal(mover, participants[1]);
   });
 
   it("identifies the indexOfMover based on the turnNum", async () => {
-    let index = await stateLib.indexOfMover.call(statePacket);
+    const index = await stateLib.indexOfMover.call(statePacket);
     // our state nonce is 15, which is odd, so it should be participant 1
     assert.equal(index, 1);
   });
 
   it("can calculate the channelId", async () => {
-    let chainId = await stateLib.channelId.call(statePacket);
-    let localId = channel.id;
+    const chainId = await stateLib.channelId.call(statePacket);
+    const localId = channel.id;
 
     assert.equal(chainId, localId);
   });
 
   it("can check if a state is signed", async () => {
     // needs to be signed by 1 as it's their move
-    let [ r, s, v ] = state.sign(participants[1]);
+    const { r, s, v } = sign(
+      state.toHex(),
+      participantB.privateKey,
+    );
+
     await stateLib.requireSignature.call(statePacket, v, r, s);
   });
 
   it("will revert if the wrong party signed", async () => {
     // needs to be signed by 1 as it's their move
-    let [ r, s, v ] = state.sign(participants[0]);
-    assertRevert(stateLib.requireSignature.call(statePacket, v, r, s));
+    const { v, r, s } = sign(state.toHex(), participants[0]);
```

Comment:
Should this be `participants[0].privateKey`?

Reply:
I think this test should fail...